.travis.yml

```
@@ -10,4 +10,11 @@ compiler:
 
 script:
   - sed -i 's|TOPCARATDIR=|TOPDIR=/home/travis/build/lbfm-rwth/carat|g' Makefile
+  - sed -i 's|CFLAGS = -g -Wall -DDIAG1|CFLAGS = -coverage -g -Wall -DDIAG1|g' Makefile
```

```suggestion
  - sed -i 's|CFLAGS = -g -Wall -DDIAG1|CFLAGS = --coverage -g -Wall -DDIAG1|g' Makefile
```
This is a rather fragile hack.. doesn't it suffice to set `CFLAGS="--coverage -g -Wall -DDIAG1" here?

Anyway, I guess once the build system is rewritten, this become moot anyway, so do whatever works quickest for you :)

---

Yes, was keeping in mind that it will be rewritten. Also, got a single `-` in `-coverage` from some docs, and it works (?). Maybe `--` is more robust for different compilers - I don't know. Also, why do you suggest to use `--coverage` permanently, or do I miss something? Anyway, that's a temporary hack, I hope...

---

So I don't know which "some docs" you are referring to, but the official GCC documentation (which I'd say is the only thing that matter here) explicitly only documents `--coverage`. See also https://gcc.gnu.org/onlinedocs/gcc-5.5.0/gcc/Debugging-Options.html#index-coverage

Moreover, I did not suggest to insert these options into the Makefile. Rather, I was saying that you should override them via the normal way to override Makefile variables: namely via arguments to`make`, not by patching the Makefile. I.e. you can do `make CFLAGS=someValue` to override the default value of that variable.

---

@fingolfin good idea - we should put `make CFLAGS=someValue` in `.travis.yml` after revising the build procedure then.